### PR TITLE
Fix "have choice" issue

### PR DESCRIPTION
**Pull request by a-lyupa**
Choose proper package name in case of multiple providers
and packages list presented

Signed-off-by: Anastasia Lyupa <a.lyupa@samsung.com>

### CODE REVIEW

File: combirepo/dependency_graph_builder.py [hunk 1]
```diff
@@ -324,10 +325,20 @@ def _search_dependencies(yum_sack, package, providers, preferables, strategy):
...
-                                                           provider,
-                                                           preferables,
-                                                           strategy)
+                    providers_in_list = 0
+                    proper_provider = None
+                    if packages_list is not None:
+                        for pr in provider:
+                            if pr.name in packages_list:
+                                providers_in_list += 1
+                                proper_provider = pr
```
Comment (ilya-palachev) [inline combirepo/dependency_graph_builder.py]: Does it mean that we prefer the last provider with the name? At each iteration, if `if` condition holds we rewrite the `proper_provider` variable. If so, it will be good to add a comment about that.

File: combirepo/dependency_graph_builder.py [hunk 2]
```diff
@@ -324,10 +325,20 @@ def _search_dependencies(yum_sack, package, providers, preferables, strategy):
                 element = Set([requirement_name])
                 provided_symbols = provided_symbols | element
                 if len(provider) != 1:
-                    provider = _handle_have_choice_problem(requirement_name,
-                                                           provider,
-                                                           preferables,
-                                                           strategy)
+                    providers_in_list = 0
```
Comment (ilya-palachev) [inline combirepo/dependency_graph_builder.py]: Not a good name for counter. Maybe `num_providers` or `providers_count` or smth. else?

File: combirepo/repository_combiner.py [hunk 3]
```diff
@@ -874,11 +876,7 @@ def generate_mic_config(output_directory_path, temporary_directory_path):
     parser.set("create", "tmpdir", mic_directory_path)
     parser.set("create", "cachedir", mic_cache_directory_path)
     parser.set("create", "outdir", output_directory_path)
-    package_manager = None
-    if rpm_patcher.developer_disable_patching:
-        package_manager = "yum"
-    else:
-        package_manager = "zypp"
+    package_manager = "zypp"
```
Comment (rayslava) [inline combirepo/repository_combiner.py:882]: Why did you remove yum?
Some distros only have yum in repository.

Comment (a-lyupa) [reply combirepo/repository_combiner.py:882]: It's temporarily removed for debug purposes.
Anyway it's not clear why --disable-rpm-patching causes yum using.

Comment (ilya-palachev) [reply combirepo/repository_combiner.py:882]: http://jira.rnd.samsung.ru/browse/TC-333

Comment (a-lyupa) [reply combirepo/repository_combiner.py:882]: > This issue can't be viewed

Comment (ilya-palachev) [reply combirepo/repository_combiner.py:882]: @rayslava ?

Comment (rayslava) [reply combirepo/repository_combiner.py:882]: @ilya-palachev  fixed already

Comment (ilya-palachev) [reply combirepo/repository_combiner.py:882]: > For now the default package manager is zypper, but it fails with strange errors ("uninstallable providers") when trying to install A sanitized and B non-sanitized when B depends on A. It seems to me that switching to yum will fix this.

If I recall correctly, there were strange problems with `zypper` and they were very hard to debug because of bindings. That's why I tried to use `yum` everywhere if possible.

But if the more modern version of `zypper` is better, you can use that instead. But it will require you some time to check that.

Comment (a-lyupa) [reply combirepo/repository_combiner.py:882]: As I see in combirepo git history zypp is used to be a default manager. It seems rather stable.

File: combirepo/dependency_graph_builder.py (hunk 1 above)
Comment (a-lyupa) [reply combirepo/dependency_graph_builder.py]: I mean to call _handle_have_choice_problem() if there are more than 1 provider in packages list.

File: combirepo/dependency_graph_builder.py (hunk 2 above)
Comment (a-lyupa) [reply combirepo/dependency_graph_builder.py]: Done

File: combirepo/repository_combiner.py (hunk 3 above)
Comment (ilya-palachev) [reply combirepo/repository_combiner.py:882]: I've no objections. Using zypp or yum is up to you.

File: combirepo/dependency_graph_builder.py (hunk 1 above)
Comment (ilya-palachev) [reply combirepo/dependency_graph_builder.py]: OK

**Review by ilya-palachev — APPROVED** [review]

**Comment by a-lyupa** [comment]
Done

File: setup.py [hunk 4]
```diff
@@ -63,7 +63,7 @@ def run(self):
 
         print "Installing man page into {0}".format(man_path)
         cmd = "bash -c 'gzip -f {0} \
-               && install -m 0644 {0}.gz {1}/'".format(man_file, man_path)
+               && install -m 0666 {0}.gz {1}/'".format(man_file, man_path)
```
Comment (rayslava) [inline setup.py]: Why do you need to create a writable man file?

File: setup.py [hunk 5]
```diff
@@ -51,7 +51,7 @@ class CustomInstallCommand(install):
     def run(self):
         install.run(self)
 
-        prefix = path.abspath(get_config_vars('prefix')[0])
+        prefix = path.abspath(get_config_vars('userbase')[0])
```
Comment (rayslava) [inline setup.py]: Doesn't this break usual system-wide install?

File: combirepo/repository_combiner.py [hunk 6]
```diff
@@ -323,14 +324,16 @@ def create_image(arch, repository_names, repository_paths, kickstart_file_path,
 
     # Now create the image using the "mic" tool:
     global mic_config_path
-    mic_command = ["sudo", "mic", "create", "loop",
+    mic_command = ["", "create", "loop",
```
Comment (rayslava) [inline combirepo/repository_combiner.py:327]: Why do you need empty first element?

**Comment by rayslava** [comment]
Please set tags to commits accordingly to changed modules.

File: combirepo/repository_combiner.py (hunk 6 above)
Comment (a-lyupa) [reply combirepo/repository_combiner.py:327]: runpy.run_path() replaces the first.

File: setup.py (hunk 5 above)
Comment (a-lyupa) [reply setup.py]: Fixed

File: setup.py (hunk 4 above)
Comment (a-lyupa) [reply setup.py]: There was an issue while removing its. Returned 0644.

**Comment by a-lyupa** [comment]
Done

**Review by rayslava — APPROVED** [review]
